Source/Core/DolphinQt2/Config/CheatWarningWidget.cpp

```diff
@@ -0,0 +1,83 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "DolphinQt2/Config/CheatWarningWidget.h"
+
+#include <QHBoxLayout>
+#include <QLabel>
+#include <QMessageBox>
+#include <QPixmap>
+#include <QPushButton>
+#include <QStyle>
+#include <QTimer>
+
+#include "Core/ConfigManager.h"
+#include "Core/Core.h"
+#include "DolphinQt2/Settings.h"
+
+CheatWarningWidget::CheatWarningWidget(const std::string& game_id) : m_game_id(game_id)
+{
+  CreateWidgets();
+  ConnectWidgets();
+
+  setHidden(true);
```

Comment:
Calling `setHidden` here and in `CreateWidgets` is redundant and a bit confusing, since it will be immediately overwritten in `Update`.